resources/table/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from "react";
+import InternalTable from "./table";
+import Pagination from "./pagination";
+
+interface Props {
+  columns: ColumnConfig<TableDataModel>[]
+  data: TableDataModel
+  isPagination?: boolean
+}
+
+const Table: React.FC<Props> = ({ columns, data, isPagination = false,  }) => {
+
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 5;
+  const totalRecords = data.length;
+  const totalPages = Math.ceil(totalRecords / itemsPerPage);
+
+  const paginatedData = !isPagination ? data : data.slice(
+    (currentPage - 1) * itemsPerPage,
+    currentPage * itemsPerPage
+  );
```

Comment:
I believe this is just to start , normally pagination is handled by the backend, you can trust that the data recived is yet paginated.

Total Records shall be a prop.

Reply:
Yeah, I am actually gonna try to make it in a way that supports frontend pagination and also it should be able to load do lazy loading where we just load single page data at a time.